### PR TITLE
GCP VM Deployment changes

### PR DESCRIPTION
**Pull request by praveenramuu**
GCP VM Deployment changes

### CODE REVIEW

File: resources/gcp/compute/Instance_vm/Instance_vm.tf [hunk 1]
```diff
@@ -0,0 +1,189 @@
+/*
```
Comment (sasikeda) [inline resources/gcp/compute/Instance_vm/Instance_vm.tf]: the directory name and filename should be changed to lower case

File: gcp_scale_templates/sub_modules/instance_template/main.tf [hunk 2]
```diff
@@ -232,21 +232,20 @@ module "storage_cluster_ingress_security_rule_using_direct_connection" {
 
 # Creates compute instances
 module "compute_cluster_instances" {
-  source                        = "../../../resources/gcp/compute/vm_instance"
+  count                         = local.cluster_type == "compute" || local.cluster_type == "combined" ? 1 : 0
+  source                        = "../../../resources/gcp/compute/vm_instance_multiple"
```
Comment (sasikeda) [inline gcp_scale_templates/sub_modules/instance_template/main.tf:236]: Please change directory layout to;
```
vm_instance_0_disk
vm_instance_multiple_disk
```

Compute must be using `vm_instance_0_disk`. and it should not take inputs
total_persistent_disks, total_local_ssd_disks, data_disk_size, data_disk_type, physical_block_size_bytes, data_disk_description, block_device_names

File: gcp_scale_templates/sub_modules/instance_template/main.tf [hunk 3]
```diff
@@ -265,19 +266,20 @@ module "compute_cluster_instances" {
 
 # Creates storage tie breaker instance
 module "storage_cluster_tie_breaker_instance" {
-  source                        = "../../../resources/gcp/compute/vm_instance"
+  count                         = local.cluster_type == "storage" || local.cluster_type == "combined" ? 1 : 0
+  source                        = "../../../resources/gcp/compute/vm_instance_multiple"
```
Comment (sasikeda) [inline gcp_scale_templates/sub_modules/instance_template/main.tf:270]: It should be using
vm_instance_multiple_disk

File: resources/gcp/compute/Instance_vm/Instance_vm.tf [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+  value = [for disk in google_compute_disk.itself : disk.self_link]
+}
+
+output "data_disk_attachment_id" {
+  value = [for disk_attach in google_compute_attached_disk.attach_data_disk : disk_attach.id]
+}
+
+output "data_disk_zone" {
+  value = [for disk in google_compute_disk.itself : disk.zone]
+}
```
Comment (sasikeda) [inline resources/gcp/compute/Instance_vm/Instance_vm.tf]: it does not look like you run `./tools/setup_dev_env.sh`

File: resources/gcp/compute/vm_instance_multiple/vm_Instances_multiple.tf [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+  availability_zones      = var.vpc_availability_zones == null ? [] : var.vpc_availability_zones
+  vpc_availability_zones  = length(local.availability_zones) > length(local.vpc_subnets) ? slice(local.availability_zones, 0, length(local.vpc_subnets)) : local.availability_zones
+  total_cluster_instances = var.total_cluster_instances == null ? 0 : var.total_cluster_instances
+  vm_configuration        = flatten([for i in range(local.total_cluster_instances) : { subnet = element(var.vpc_subnets, i), zone = element(local.vpc_availability_zones, i), vm_name = "${var.instance_name_prefix}-${i}" }])
+}
+
+# Creating multiple instances
+module "instances_multiple" {
+  count                         = length(local.vm_configuration)
+  source                        = "../Instance_vm"
```
Comment (sasikeda) [inline resources/gcp/compute/vm_instance_multiple/vm_Instances_multiple.tf]: Don't refer back to vm_instance, 
you can copy the entire code here.

File: resources/gcp/compute/Instance_vm/Instance_vm.tf [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+
+  dynamic "scratch_disk" {
+    for_each = range(var.total_local_ssd_disks)
+    content {
+      interface = "NVME"
+    }
+  }
+
+  lifecycle {
+    ignore_changes = [attached_disk, metadata_startup_script]
```
Comment (sasikeda) [inline resources/gcp/compute/Instance_vm/Instance_vm.tf]: lifecycle `attached_disk` is not required here.
To use it you will need to use dynamic attached_block

File: resources/gcp/compute/Instance_vm/Instance_vm.tf (hunk 4 above)
Comment (praveenramuu) [reply resources/gcp/compute/Instance_vm/Instance_vm.tf]: Done sasi

File: gcp_scale_templates/sub_modules/instance_template/main.tf (hunk 2 above)
Comment (praveenramuu) [reply gcp_scale_templates/sub_modules/instance_template/main.tf:236]: Thanks sasi, will take care in next PR

**Pull request merged**